Description: Disable C compiler warnings entirely
 In Meson 1.4.0, warnings for Vala-transpiled C code were unhidden, which
 causes the Budgie build to turn into unreadable logspam due to the
 absolutely abysmal C code generated by Vala. This PR restores the prior
 behavior by disabling all warnings for C code.
 based on PR https://github.com/BuddiesOfBudgie/budgie-desktop/pull/555
Author: David Mohammed <fossfreedom@ubuntu.com>
Forwarded: not-needed
Last-Update: 2024-09-30

--- budgie-desktop-10.8.2.orig/meson.build
+++ budgie-desktop-10.8.2/meson.build
@@ -12,6 +12,17 @@ project(
     ],
 )
 
+# Vala generates bad C code and missing these on gcc 14 will cause FTBFS
+# Additionally, Meson 1.4 unhides warnings from valac-generated C code,
+# which causes unreadable logspam. Reenables prior behavior.
+am_cflags = [
+    '-w',
+    '-Wno-incompatible-pointer-types',
+    '-Wno-implicit-function-declaration',
+]
+add_global_arguments(am_cflags, language: 'c')
+
+
 budgie_screensaver  = find_program('budgie-screensaver', required: false)
 gnome_screensaver = find_program('gnome-screensaver', required: false)
 with_gnome_screensaver = get_option('with-gnome-screensaver')
